src/main/java/gregtech/common/blocks/BlockFrame.java

```diff
@@ -185,6 +186,24 @@ public boolean replaceWithFramedPipe(World worldIn, BlockPos pos, IBlockState st
         return false;
     }
 
+    public boolean removeFrame(World world, BlockPos pos, IBlockState state, EntityPlayer player, ItemStack stack, EnumFacing facing) {
+
+        TileEntity te = world.getTileEntity(pos);
+        if (te instanceof IPipeTile && ((IPipeTile<?, ?>) te).getFrameMaterial() != null) {
+            TileEntityPipeBase<?, ?> pipeTile = (TileEntityPipeBase<?, ?>) te;
```

Comment:
Here is potential bug waiting. Right now it won't happened as we don't have any other `IPipeTile` implementation which are not of type `TileEntityPipeBase` and could be here. But if there was one then `if` check would be `true` here but casting will return `null`. Leading to null reference expectation. So you should check for same value as you are going to cast to.